diff --git a/token.h b/token.h
index 95adf01..823ecf8 100644
--- a/token.h
+++ b/token.h
@@ -128,8 +128,8 @@ IFILE {
 
 #define TOKENSTREAM struct _tokenstream
 TOKENSTREAM {
-	TOKEN token[32];		// 32 ought to be enough for anybody (including XiA!)
-	char * string[32];		// same for attached strings
+	TOKEN token[64];		// 32 ought to be enough for anybody (including XiA!)
+	char * string[64];		// same for attached strings
 };
 
 // Information about a macro invocation
